web-api/src/storage.rs

```diff
@@ -71,6 +71,12 @@ pub(crate) enum DeletionError {
     PartialFailure { errors: Vec<DocumentIdAsObject> },
 }
 
+impl From<sqlx::Error> for DeletionError {
+    fn from(e: sqlx::Error) -> Self {
+        DeletionError::General(e.into())
+    }
+}
```

Comment:
:art: we don't have to impl this conversion manually, instead we can annotate the variants of the `DeletionError` like
```rust
#[derive(Debug, From)]
pub(crate) enum DeletionError {
    #[from(types(sqlx::Error))]
    General(Error),
    #[from]
    PartialFailure { errors: Vec<DocumentIdAsObject> },
}
```

Reply:
Nice, thanks